lib/vehicle_factory.rb

```diff
@@ -0,0 +1,59 @@
+class VehicleFactory
+  attr_reader :created_vehicles
+
+  def initialize
+    @created_vehicles = []
+  end
+
+  def create_vehicles(state_registration)
+    @created_vehicles = state_registration.map do |vehicle|
+      Vehicle.new(
+        vin: vehicle[:vin_1_10], 
+        year: vehicle[:model_year], 
+        make: vehicle[:make], 
+        model: vehicle[:model], 
+        engine: :ev)
+    end 
+  end
+
```

Comment:
your logic in this method looks good. There was no need for an attribute, since we only cared about teh return value of this `create_vehicles` method. This return value might change depending on what the API returns, so that's why we don't need to store the state in an attribute, and we can focus on just returning the vehicles that were created when the `create_vehicles()` method was called